### ISSUE TITLE
[Sentry] ModuleIntegration wasn't able to extract modules: module integration failed

### ISSUE THREAD

**Issue by parasssh** [getsentry/sentry-go#197]
https://github.com/dgraph-io/dgraph/issues/5166 

A user is seeing this log message from Sentry when using our application. More details at the link above.
`[Sentry] ModuleIntegration wasn't able to extract modules: module integration failed`

Questions:
1. What does this message mean?
2. Under what circumstances do we see this? 
3. And is it a concern or can be ignored?

FWIW, this user's sentry event with ID `cc19eb68a1de4cdc8f4808283e9d17ea` was indeed reported on our Sentry Dashboard.

**Comment by parasssh** [comment]
The sentry-go SDK is version `0.5.1`

**Comment by rhcarvalho** [comment]
Hi @parasssh! Thanks for bringing this to our attention.

> 1\. What does this message mean?

It means that the SDK could not get module information to attach to the error prior to reporting.

> 2\. Under what circumstances do we see this?

It will happen whenever the SDK doesn't find a `go.mod` file or `vendor/modules.txt` or `vendor/vendor.json`.

<details>
<summary>Current implementation</summary>

https://github.com/getsentry/sentry-go/blob/81d3dd83cbe8542a9103bd5bb72c42357dfbc768/integrations.go#L42-L68
</details>

> 3\. And is it a concern or can be ignored?

It can be totally ignored. The only consequence is that the event in Sentry will not include a list of modules/packages.

In case of `dgraph`, which AFAIK is a program rather than a library, I'd say it is not so critical to have a list of packages listed in Sentry, since the dependencies for a given binary are known and do not change in runtime.

---

Whenever we fix #184, we'll likely be able to get a list of modules from the running binary itself, without a dependency on `go.mod` (via https://golang.org/pkg/runtime/debug/#BuildInfo).

In the meantime, if the message is bothering and you'd like to get rid of it right now, one way to do so is to disable the Modules Integration:

```go
func main() {
	err := sentry.Init(sentry.ClientOptions{
		// ...
		Integrations: func(integrations []sentry.Integration) []sentry.Integration {
			filtered := integrations[:0]
			for _, integration := range integrations {
				if integration.Name() != "Modules" {
					filtered = append(filtered, integration)
				}
			}
			return filtered
		},
	})
	// ...
}
```
